docs/icicle/primitives/poseidon.md

```diff
@@ -0,0 +1,233 @@
+# Poseidon
+
+Poseidon hash is a popular hash in the ZK ecosystem primarily because its optimized to work over large prime fields, a common setting for ZK proofs, thereby minimizing the number of multiplicative operations required.
+
+Poseidon hash function has also been specifically designed to be efficient when implemented within ZK circuits, Poseidon hash uses far less constraints compared to other hash functions like Keccak or SHA-256 in the context of ZK circuits.
```

Comment:
I think it makes sense to refer to Poseidon as strictly "Poseidon" after the first sentence describing it as a hash function